common/djangoapps/track/views/__init__.py

```diff
@@ -111,27 +111,14 @@ def server_track(request, event_type, event, page=None):
     except:
         username = "anonymous"
 
-    # define output:
-    event = {
-        "username": username,
-        "ip": _get_request_ip(request),
-        "referer": _get_request_header(request, 'HTTP_REFERER'),
-        "accept_language": _get_request_header(request, 'HTTP_ACCEPT_LANGUAGE'),
-        "event_source": "server",
-        "event_type": event_type,
-        "event": event,
-        "agent": _get_request_header(request, 'HTTP_USER_AGENT').encode().decode('latin1'),
-        "page": page,
-        "time": datetime.datetime.utcnow().replace(tzinfo=pytz.utc),
-        "host": _get_request_header(request, 'SERVER_NAME'),
-        "context": eventtracker.get_tracker().resolve_context(),
-    }
-
-    # Some duplicated fields are passed into event-tracking via the context by track.middleware.
-    # Remove them from the event here since they are captured elsewhere.
-    shim.remove_shim_context(event)
-
-    log_event(event)
+    context_override = contexts.course_context_from_url(page)
+    context_override['username'] = username
+    context_override['event_source'] = 'server'
+    context_override['page'] = page
+
+    event_tracker = eventtracker.get_tracker()
```

Comment:
@danialmalik we are neither using `server_track` nor `task_track` according to Fahad's PR. Then why need these changes?

Reply:
Previously we just replaced the `server_track` and `task_track` calls with eventtracking's `tracker.emit` calls. But upon observing [this PR](https://github.com/edx/edx-platform/pull/7802/files#diff-e0483519f30846bafc1b92fb53f9aee5R325), we realized that we can follow the same steps and replace the `log_event` (which uses `tracker.send`) with eventtracking's call. This way we won't have to change EVERY server_track and task_track call.